pkg/util/secret_test.go

```diff
@@ -0,0 +1,244 @@
+package util
+
+import (
+	"errors"
+	"reflect"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+	coretesting "k8s.io/client-go/testing"
+)
+
+func TestCreateOrUpdateSecret(t *testing.T) {
+	type args struct {
+		client kubernetes.Interface
+		secret *corev1.Secret
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "create success",
+			args: args{
+				client: fake.NewSimpleClientset(),
+				secret: makeSecret("test"),
+			},
+			wantErr: false,
+		},
+		{
+			name: "update success",
+			args: args{
+				client: fake.NewSimpleClientset(makeSecret("test")),
+				secret: makeSecret("test"),
+			},
```

Comment:
I think there is a problem with the `CreateOrUpdate` function, as I mentioned on https://github.com/karmada-io/karmada/pull/2596#discussion_r1006333298.
I'm surprised the test can't find it.

Reply:
fake client doesn't check the resourceVersion.

https://github.com/karmada-io/karmada/blob/531d6bf2acbe9dd572ef5f09437ad713bc2f41e1/vendor/k8s.io/client-go/testing/fixture.go#L116-L126

Reply:
OK.